tardis/apps/publication_workflow/static/publication-form/publication_form.js

```diff
@@ -0,0 +1,504 @@
+            {
+                'name': '',
+                'institution':'',
+                'email': ''
+            }
+        );
+    };
+
+    // Remove author from publication
+    vm.removeAuthorEntry = function (idx) {
+        vm.formData.authors.splice(idx, 1);
+    };
+
+    // Copy acknowledgement text to acknowledgement field
+    vm.copyAcknowledgement = function (text) {
+        if (vm.formData.acknowledgements.indexOf(text) === -1) {
+            if (vm.formData.acknowledgements.length > 0) {
+                vm.formData.acknowledgements += " ";
+            }
+            vm.formData.acknowledgements += text;
+        }
+    };
+
+    // pubFormCtrl is the name of the controller instance used in templates/form.html
+    $scope.$watch('pubFormCtrl.formData.selectedLicenseId', function (newVal, _oldVal) {
+        if (angular.isDefined(vm.formData.licenses)) {
+            for (var i in vm.formData.licenses) {
+                if (vm.formData.licenses.hasOwnProperty(i)) {
+                    var license = vm.formData.licenses[i];
+                    if (license.id === newVal) {
+                        vm.formData.selectedLicense = license;
+                        break;
+                    }
+                }
+            }
+        }
+    });
+
+    vm.saveAndClose = function () {
+        if (!vm.formData.action) {
+            // vm.formData.action will be blank if user clicked "Save and finish later"
+            if (vm.currentPage.title == 'Select datasets') {
+                vm.formData.action = "update-dataset-selection";
+            } else if (vm.currentPage.title == 'Extra information') {
+                vm.formData.action = "update-extra-info";
+            } else if (vm.currentPage.title == 'Attribution and licensing') {
+                vm.formData.action = 'update-attribution-and-licensing';
+            }
+        }
+        angular.element($document[0].querySelector('.ngdialog')).scrollTop(0);
+        saveFormState(
+            function () { // On success
+                // Preventing using $scope directly silences ESLint's angular/controller-as error:
+                var dialogScope = $scope;
+                dialogScope.closeThisDialog();
+            },
+            function () {
+            } // On error
+        );
+    };
+
+
+    vm.setEmbargoToToday = function () {
+        vm.formData.embargo = new Date();
+    };
+    // Ensures that the embargo date is a Date object
+    // pubFormCtrl is the name of the controller instance used in templates/form.html
+    $scope.$watch('pubFormCtrl.formData.embargo', function (newVal, _oldVal) {
+        if (angular.isString(newVal)) {
+            vm.formData.embargo = new Date(newVal);
+        }
+    });
+
+    vm.initDatasetExtraInfo = function(formIndex, datasetIndex, datasetDescription) {
+         if (angular.isUndefined(vm.formData.extraInfo[formIndex+'.'+datasetIndex])) {
+             vm.formData.extraInfo[formIndex+'.'+datasetIndex] = {};
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933951)
```suggestion
             vm.formData.extraInfo[formIndex+"."+datasetIndex] = {};
```